pkg/app/piped/eventwatcher/eventwatcher.go

```diff
@@ -314,79 +329,81 @@ func (w *watcher) execute(ctx context.Context, repo git.Repo, repoID string, eve
 		gitUpdateEvent   = false
 	)
 	for _, e := range eventCfgs {
-		var (
-			matcher = e.Matcher
-			handler = e.Handler
-		)
-		notHandledEvents := w.eventLister.ListNotHandled(matcher.Name, matcher.Labels, milestone+1, numToMakeOutdated)
-		if len(notHandledEvents) == 0 {
-			continue
-		}
-		if len(notHandledEvents) > 1 {
-			// Events other than the latest will be OUTDATED.
-			for _, e := range notHandledEvents[1:] {
-				outDatedEvents = append(outDatedEvents, &pipedservice.ReportEventStatusesRequest_Event{
-					Id:                e.Id,
-					Status:            model.EventStatus_EVENT_OUTDATED,
-					StatusDescription: fmt.Sprintf("The new event %q has been created", notHandledEvents[0].Id),
-				})
+		for _, cfg := range e.Configs {
+			var (
+				matcher = cfg.Matcher
+				handler = cfg.Handler
+			)
+			notHandledEvents := w.eventLister.ListNotHandled(matcher.Name, matcher.Labels, milestone+1, numToMakeOutdated)
```

Comment:
I start feeling that this RPC is being called quite frequently.
In the future, maybe we need to add a new RPC to check multiple events simultaneously.

Reply:
Exactly, but it's not so frequently in the current spec because the count of `eventCfgs` is equal to the count of apps and the count of `e.Config` is almost to be one except for edge cases.
So Let's add a new RPC when we add a new event type.